tests/test_dds_creation.py

```diff
@@ -0,0 +1,88 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from pydeseq2.dds import DeseqDataSet
+from pydeseq2.utils import build_design_matrix
+from pydeseq2.utils import load_example_data
+
+
+@pytest.mark.parametrize(
+    "design_factors",
+    [
+        ["condition"],
+        ["condition", "group"],
+        ["condition:group"],
+        ["condition", "group", "condition:group"],
+        "~condition:group",
+        "~condition+condition:group",
+    ],
+)
```

Comment:
Relatedly to one of my previous comments, we might want to add a test with formulaic and a design factor containing an underscore in its name.

Reply:
There is one already as everything ends up on formulaic